src/Dom/Document.php

```diff
@@ -536,6 +537,23 @@ private function move_invalid_head_nodes_to_body() {
 		}
 	}
 
+	/**
+	 * Converts a possible head[profile] attribute to link[rel=profile].
+	 *
+	 * The head[profile] attribute is only valid in HTML4, not HTML5.
+	 * So if it exists, add it to the <head> as a link[rel=profile] and strip the attribute.a -
+	 */
+	private function convert_head_profile_to_link() {
+		$profile = $this->head->getAttribute( 'profile' );
+		if ( $profile ) {
+			$link = $this->createElement( 'link' );
+			$link->setAttribute( 'rel', 'profile' );
+			$link->setAttribute( 'href', $profile );
+			$this->head->appendChild( $link );
+			$this->head->removeAttribute( 'profile' );
+		}
```

Comment:
Only thing left is to make sure that a falsy `profile` attribute is removed, so like:

```php
if ( ! $this->head->hasAttribute( 'profile' ) ) {
	return;
}

$profile = $this->head->getAttribute( 'profile' );
if ( $profile ) {
	$link = $this->createElement( 'link' );
	$link->setAttribute( 'rel', 'profile' );
	$link->setAttribute( 'href', $profile );
	$this->head->appendChild( $link );
}

$this->head->removeAttribute( 'profile' );
```

Reply:
Sounds good, that's applied in https://github.com/ampproject/amp-wp/pull/4193/commits/6ebae07a5a2f804b73eb465120e41f06259ec2bd